src/core/ext/filters/client_channel/lb_policy/xds/xds_load_balancer_api.cc

```diff
@@ -18,12 +18,129 @@
 
 #include <grpc/support/port_platform.h>
 
-#include "pb_decode.h"
-#include "pb_encode.h"
 #include "src/core/ext/filters/client_channel/lb_policy/xds/xds_load_balancer_api.h"
 
 #include <grpc/support/alloc.h>
 
+namespace grpc_core {
+
+namespace {
+
+constexpr char kEdsTypeUrl[] =
+    "type.googleapis.com/envoy.api.v2.ClusterLoadAssignment";
+constexpr char kEndpointRequired[] = "endpointRequired";
+upb_alloc* g_arena = nullptr;
+
+void InitArena() { g_arena = upb_arena_new(); }
+
+upb_alloc* arena() {
+  static gpr_once once = GPR_ONCE_INIT;
+  gpr_once_init(&once, InitArena);
+  return g_arena;
+}
+
+}  // namespace
+
+XdsDiscoveryRequest* XdsRequestCreate(const char* service_name) {
+  XdsDiscoveryRequest* request = envoy_api_v2_DiscoveryRequest_new(arena());
+  XdsNode* node = envoy_api_v2_DiscoveryRequest_mutable_node(request, arena());
+  XdsStruct* metadata = envoy_api_v2_core_Node_mutable_metadata(node, arena());
+  XdsFieldsEntry* field = google_protobuf_Struct_add_fields(metadata, arena());
```

Comment:
Ack, this is unfortunate. Your first code went for an API that is going to change soon. :)

For maps the API needs to be more like insert(key, value). So this is the correct code for now, but it will need to change soon.

Reply:
I see. I will change this when I update to use the new API. Thanks!